### PR TITLE
Provider runtime options

### PR DESCRIPTION
**Pull request by amwat**
follow up to https://github.com/kubernetes-sigs/kind/pull/1341

### CODE REVIEW

File: pkg/internal/runtime/runtime.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+	"sigs.k8s.io/kind/pkg/log"
+)
+
+// GetDefault selected the default runtime from the environment override
+func GetDefault(logger log.Logger) cluster.ProviderOption {
+	switch p := os.Getenv("KIND_EXPERIMENTAL_PROVIDER"); p {
+	case "":
+		return nil
+	case "podman":
+		logger.Warn("using runtime \"podman\" due to KIND_EXPERIMENTAL_PROVIDER")
```
Comment (BenTheElder) [inline pkg/internal/runtime/runtime.go]:
```suggestion
		logger.Warn("using podman due to KIND_EXPERIMENTAL_PROVIDER")
```

File: pkg/internal/runtime/runtime.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+		return cluster.ProviderWithPodman()
+	case "docker":
+		logger.Warn("using runtime \"docker\" due to KIND_EXPERIMENTAL_PROVIDER")
```
Comment (BenTheElder) [inline pkg/internal/runtime/runtime.go]:
```suggestion
		logger.Warn("using docker due to KIND_EXPERIMENTAL_PROVIDER")
```

File: pkg/internal/runtime/runtime.go [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+		return cluster.ProviderWithDocker()
+	default:
+		logger.Warnf("ignoring unknown runtime %q from KIND_EXPERIMENTAL_PROVIDER", p)
```
Comment (BenTheElder) [inline pkg/internal/runtime/runtime.go]:
```suggestion
		logger.Warnf("ignoring unknown value %q for KIND_EXPERIMENTAL_PROVIDER", p)
```

**Comment by BenTheElder** [comment]
looks great! painting the bikeshed

**Comment by amwat** [comment]
hmm https://github.com/kubernetes-sigs/kind/issues/999 strikes back?

https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kind/1345/pull-kind-e2e-kubernetes/1230661253223944192#1:build-log.txt%3A58630

will check

**Pull request merged**